src/Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs

```diff
@@ -225,17 +253,37 @@ public static class WebApiServiceCollectionExtensions
         /// </summary>
         /// <param name="services">Service collection to which to add authentication</param>
         /// <param name="configuration">Configuration</param>
+        /// <param name="configSectionName">Section name in the config file (by default "AzureAD")</param>
+        /// <param name="jwtBearerScheme">Scheme for the JwtBearer token</param>
         /// <returns></returns>
         public static IServiceCollection AddProtectedWebApiCallsProtectedWebApi(
             this IServiceCollection services,
             IConfiguration configuration,
             string configSectionName = "AzureAd",
             string jwtBearerScheme = JwtBearerDefaults.AuthenticationScheme)
+        {
+            return services.AddProtectedWebApiCallsProtectedWebApi(
+                jwtBearerScheme,
+                options => configuration.Bind(configSectionName, options),
+                options => configuration.Bind(configSectionName, options));
+        }
+        /// <summary>
+        /// Protects the Web API with Microsoft identity platform (formerly Azure AD v2.0)
+        /// This supposes that the configuration files have a section named configSectionName (typically "AzureAD")
+        /// </summary>
+        /// <param name="services">Service collection to which to add authentication</param>
+        /// <param name="configuration">Configuration</param>
+        /// <returns></returns>
+        public static IServiceCollection AddProtectedWebApiCallsProtectedWebApi(
+            this IServiceCollection services,
+            string jwtBearerScheme,
+            Action<ConfidentialClientApplicationOptions> msalAppOptionsMapper,
+            Action<MicrosoftIdentityOptions> microsoftIdentityOptionsMapper)
```

Comment:
Should these two Action parameters be renamed? The user can pass any action, so I think `*Mapper` is not generic enough suffix. The parameter name for the ASP.NET `Configure` method below is `configureOptions`. We could rename these to `configureMsalAppOptions` and `configureMicrosoftIdentityOptions` for consistency.

Reply:
Done!